include/qpdf/QPDFObjectHandle.hh

```diff
@@ -1573,7 +1573,7 @@ class QPDFObjectHandle
     void typeWarning(char const* expected_type, std::string const& warning);
     void objectWarning(std::string const& warning);
     void assertType(char const* type_name, bool istype);
-    void dereference();
+    bool dereference(bool unguarded = false);
```

Comment:
See impl for comment -- change `unguarded` to `allow_uninitialized` and remove default.